### PR TITLE
Add kernel config test

### LINKED ISSUE #751: Include a test that checks presence/absence of kernel `CONFIG_` options
**What would you like to be added**:

Have a test in the Garden Linux build that checks if the kernel has been compiled with certain `CONFIG_` options. It should be possible to specify options, that need to be set, that must not be set or that must have a specific value in a file and the test compares the desired values in said file to the actual values the kernel was compiled with.

**Why is this needed**:
With Garden Linux 576 we found that several patch releases had Kernels that were compiled fundamentally differently to previous versions which broke the workloads of users on Gardener. Once we release a Garden Linux release, we need to be confident that critical kernel parameters do not change unnoticed in subsequent patch releases.

### PR DESCRIPTION
**Pull request by chrinorse**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
add a pytest to test if kernel parameters are set or not set as expected
**Which issue(s) this PR fixes**:
Fixes #751 

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Review by MrBatschner — CHANGES_REQUESTED** [review]

**Comment by Vincinator** [comment]
I am working on a kernel config test for the gitlab pipeline. This gitlab version checks that all configs are included before deploying the package in the repo.gardenlinux.io apt repository.

However, +1 for this additional config test here in the gardenlinux repo. 

Here are a few notes that may help for this PR:

- In the linux-image repo we have a config file, where we specify the gardenlinux kernel configs. 
https://gitlab.com/gardenlinux/gardenlinux-package-linux-5.10/-/blob/main/debian/config/config.gardenlinux
    - Heads Up: There are old intel DFL driver flags. Not sure if we want to keep them for 5.10, since some of the flags can not be included in the final image because they do not exist or depend on something that we must explicitly enable first. Therefore the config tests fail.
    - I will implement a skip-test comment that can be added in the config.gardenlinux. This will mark the next config flag after the skip-test comment to be ignored by the test. We need to keep this in mind before we sync the config.gardenlinux file with the `features/base/test/kernel-config.d/kernel-config.txt`
- Reasons to test if the desired configs are actually included in the image:
    - specified config does not exist in this kernel version
    - bug (e.g. CONFIG_HYPERV_UTILS in `5.10.110`, fixed/reverted again in `5.10.111`)

File: tests/helper/tests/kernel_config.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            expected_config = utils.read_test_config(
+                enabled_features, 'kernel-config', '.txt',
+                filter_comments = False)
+
+            config_dict = cls._to_dict(cls.instance, output)
+            expected_config_dict = cls._to_dict(cls.instance, expected_config)
+
+            logger.info(expected_config_dict)
+            not_matching = []
+            for key, value in expected_config_dict.items():
```
[the remark this answers is not in the document]
Comment (chrinorse) [reply tests/helper/tests/kernel_config.py:66]: Good point. Added a check to treat 'n' and 'is not set' as equal.

**Review by MrBatschner — CHANGES_REQUESTED** [review]

File: tests/helper/tests/kernel_config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrinorse) [reply tests/helper/tests/kernel_config.py:67]: yes, added stripping of white space characters from the value. Leading and trailing space, tab and newline are removed now.

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Pull request merged**